tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(not status["runner"]["selected"], "lag member is still up")
+
+
+def test_peer_retry_count_disabled(duthost, nbrhosts, higher_retry_count_on_peers, disable_retry_count_on_peer):
+    """
+    Test that peers reset the retry count to 3 when the feature is disabled
+    """
+
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+    port_channels = cfg_facts["PORTCHANNEL"].keys()
+    for port_channel in port_channels:
+        port_channel_status = duthost.get_port_channel_status(port_channel)
+        for _, status in list(port_channel_status["ports"].items()):
+            pytest_assert(status["runner"]["partner_retry_count"] == 3,
+                          "partner retry count is incorrect; expected 3, but is {}"
+                          .format(status["runner"]["partner_retry_count"]))
+
+    for nbr in list(nbrhosts.keys()):
+        processRc = nbrhosts[nbr]['host'].shell("sudo config portchannel retry-count get PortChannel1",
+                                                module_ignore_errors=True)
+        pytest_assert(processRc["failed"], "Expected failure for getting retry count, but instead it succeeded")
+
+
+class TestDutRetryCount:
+    def test_retry_count(self, duthost, nbrhosts, higher_retry_count_on_dut):
+        """
+        Test that peers see new retry count when DUT updates retry count.
+        """
+        cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+        port_channels = cfg_facts["PORTCHANNEL"].keys()
+        for port_channel in port_channels:
+            port_channel_status = duthost.get_port_channel_status(port_channel)
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5,
+                          "retry count on DUT is incorrect; expected 5, but is {}"
+                          .format(port_channel_status["runner"]["retry_count"]))
+
+        for nbr in list(nbrhosts.keys()):
+            port_channel_status = nbrhosts[nbr]['host'].get_port_channel_status("PortChannel1")
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(status["runner"]["selected"], "lag member is not up")
+                pytest_assert(status["runner"]["partner_retry_count"] == 5,
+                              "partner retry count is incorrect; expected 5, but is {}"
+                              .format(status["runner"]["partner_retry_count"]))
+
+    def test_retry_count_packet_version(self, duthost, nbrhosts, higher_retry_count_on_dut, scapy_lacp_layer):
+        """
+        Test that peers and DUT use new LACPDU version when DUT uses a non-standard retry count
+        """
+        check_lacpdu_packet_version(duthost)
+
+    def test_kill_teamd_peer_lag_up(self, duthost, nbrhosts, higher_retry_count_on_peers, config_reload_on_cleanup):
```

Comment:
Same comments here as for `test_kill_teamd_lag_up`

Reply:
Currently, teamd itself doesn't have any connection to our redis DB; the only indication of warm reboot it gets is if it gets either a SIGUSR1 or SIGUSR2. I would rather not have it determine whether the system will be undergoing warm reboot based solely on that.

There is, however, a teammgrd process and teamsyncd process running in the teamd docker that do have connections to redis DB (I forget what the purpose of each of those daemons are). If we want this feature, one possibility might be to have those processes monitor the warm-reboot flag, and when it changes, set it in teamd via `teamdctl state item set ...`. That does increase the complexity of the protocol logic, though (if it's been more than 90 seconds already, and the warm reboot flag gets cleared, then should it immediately bring the LAGs down?).